crates/polars-core/src/series/from.rs

```diff
@@ -84,9 +84,12 @@ impl Series {
             #[cfg(feature = "dtype-categorical")]
             dt @ (Categorical(rev_map, ordering) | Enum(rev_map, ordering)) => {
                 let cats = UInt32Chunked::from_chunks(name, chunks);
+                let rev_map = rev_map
+                    .clone()
+                    .unwrap_or_else(|| Arc::new(RevMapping::default()));
```

Comment:
We should assert `cats` is emtpy if we set an empty rev map.

Reply:
Maybe should add it directly in `from_cats_and_rev_map_unchecked` under cfg(debug_assertions)? 
But this seems to be the only place when they're unwrapped, so I added it just here for now.